pkg/controller/operators/olm/operator.go

```diff
@@ -1416,6 +1416,73 @@ func (a *Operator) apiServiceOwnerConflicts(csv *v1alpha1.ClusterServiceVersion)
 	return nil
 }
 
+func (a *Operator) apiServiceOwnerConflictsNew(csv *v1alpha1.ClusterServiceVersion) error {
+	// Get replacing CSV if exists
+	replacing, err := a.lister.OperatorsV1alpha1().ClusterServiceVersionLister().ClusterServiceVersions(csv.GetNamespace()).Get(csv.Spec.Replaces)
+	if err != nil && !k8serrors.IsNotFound(err) && !k8serrors.IsGone(err) {
+		return err
+	}
+
+	owners := []ownerutil.Owner{csv}
+	if replacing != nil {
+		owners = append(owners, replacing)
+	}	
+	
+	for _, desc := range csv.GetOwnedAPIServiceDescriptions() {
+		// Check if the APIService exists
+		apiService, err := a.lister.APIRegistrationV1().APIServiceLister().Get(desc.GetName())
+		if err != nil && !k8serrors.IsNotFound(err) && !k8serrors.IsGone(err) {
+			return err
+		}
+
+		if apiService == nil {
+			continue
+		}
+
+		labels := apiService.GetLabels()
+		ownerKind := labels[ownerutil.OwnerKind]
+		ownerName := labels[ownerutil.OwnerKey]
+		ownerNamespace := labels[ownerutil.OwnerNamespaceKey]
+
+		if ownerKind == "" || ownerNamespace == "" || ownerName == "" {
+			// If owner is set, do we always expect all three labels to be set?
+			// Question: Can a cluster scoped object ever own an APIService?
+			return ErrAPIServiceOwnerConflict
+		}
+
+		// Is it owned by this csv or the one it replaces.
+		if ownerutil.AdoptableLabels(apiService.GetLabels(), true, owners...) {
```

Comment:
If you check the owner for existence before this, and add it to the `owners` slice _if it's not found_, then you should be able to get rid of the rest of this function.